server/datastore/mysql/software_installers.go

```diff
@@ -1073,35 +1073,77 @@ WHERE
 
 func (ds *Datastore) GetSummaryHostSoftwareInstalls(ctx context.Context, installerID uint) (*fleet.SoftwareInstallerStatusSummary, error) {
 	var dest fleet.SoftwareInstallerStatusSummary
+	// TODO(uniq): do we need to support team filtering or is that not needed because installers are
+	// associated to teams?
 
-	// TODO(mna): must also look in upcoming queue for pending, and most recent
-	// attempt for an installer might be in upcoming queue...
-	stmt := `
+	// TODO(uniq): do we need to handle the host_deleted_at and removed conditions somehow for upcoming
+	// installs like is happening with past installs?
```

Comment:
My understanding is that those fields matter ONLY for installs in a final state (success or failed), NOT for pending states. That's because they are used to filter out what can or cannot be installed (set to `removed` when the software is deleted from the host), and to fix stats. It's not relevant for the upcoming activities because the rows get deleted if a host gets deleted, and pending installs get deleted if an installer gets removed.

Reply:
Thanks! That lines up with my understanding too.